src/vm/amd64/JitHelpers_InlineGetThread.asm

```diff
@@ -856,6 +856,9 @@ NESTED_ENTRY JIT_MonTryEnter_InlineGetThread, _TEXT
         mov                 [rsp + 8h], rcx
         .savereg            rdx, 10h
         mov                 [rsp + 10h], rdx
+	
+        push_nonvol_reg     r12 ; rcx now at [rsp + 10h]
+		
```

Comment:
If doing the `cmp` directly, I don't think I need to restore `rdx`  before exit? (Its not done in the thinlock)